api/base/views.py

```diff
@@ -437,7 +453,71 @@ def root(request, format=None, **kwargs):
     if utils.has_admin_scope(request):
         return_val['meta']['admin'] = True
 
-    return Response(return_val)
+    resp = Response(return_val)
+
+    for key, value in cookies.items():
+        resp.set_cookie(key, value)
+
+    return resp
+
+
+def sloan_study_disambiguation(request):
+    """
+    This is a hack to set flags and cookies for out Sloan study, it can be deleted when the study is complete.
+    """
+    user = request.user
+    cookies = {}
+
+    flags = []
+    sloan_data = {}
+    for flag in Flag.objects.all():
+        active = flag.is_active(request._request)
+        if flag.name in SLOAN_FLAGS and flag.everyone is None:
+            sloan_data[flag.name] = active
+
+        # User tags should override any cookie info
+        if user and not user.is_anonymous:
+            if user.all_tags.filter(name=flag.name):
+                sloan_data[flag.name] = True
+                active = True
+            if user.all_tags.filter(name=f'no_{flag.name}'):
+                sloan_data[flag.name] = False
+                active = False
+
+        if active:
+            flags.append(flag.name)
+
+    # This is used exclusively for our partnership with Sloan and can be deleted after their study is complete.
+    referer_url = request.environ.get('HTTP_REFERER', '')
+    provider_domains = list(PreprintProvider.objects.exclude(domain='').values_list('domain', flat=True))
+
+    provider_domains = [domains for i, domains in enumerate(provider_domains) if referer_url.startswith(domains)]
```

Comment:
```suggestion
    provider_domains = [domain for domain in provider_domains if referer_url.startswith(domain)]
```

Reply:
Just vestigial, I wanted to see if the string was in the string in the list not if the string was in the list, but that a obvious no-no when startswith also checks where in the string the fragment is contained.